src/components/input/input.ts

```diff
@@ -316,7 +316,9 @@ export default class SlInput extends ShoelaceElement {
     // submitting to allow users to cancel the keydown event if they need to
     if (event.key === 'Enter' && !hasModifier) {
       setTimeout(() => {
-        if (!event.defaultPrevented) {
+        // isComposing is true when enter key is pressed while choosing the chinese/japanese/etc character
+        // so this check will prevent form submitting after choosing character by enter key pressing
+        if (!event.defaultPrevented && !event.isComposing) {
```

Comment:
Should we be using `!event.defaultPrevented && event.eventPhase === eventEvent.AT_TARGET` instead? This would ensure that the event was fired from the target, which seems a bit more intuitive from a maintenance perspective.

Link for reference: https://developer.mozilla.org/en-US/docs/Web/API/Event/eventPhase

Reply:
Bumping this in case it was missed. If you can provide a repro for me to follow, I'm happy to test the alternate syntax myself.

Reply:
hey, sorry for late response. Your approach looks good to me. Should I commit the changes?

Reply:
Sounds good. Thanks!

Reply:
I've just double checked your approach and it actually doesn't work. The `event.eventPhase` is always `0` so it doesn't submit the form at all

Reply:
Thanks for verifying. I'm OK with it as-is then. We can improve it as needed later on.